controllers/gce/instances/fakes.go

```diff
@@ -145,10 +148,23 @@ func (f *FakeInstanceGroups) RemoveInstancesFromInstanceGroup(name, zone string,
 	return nil
 }
 
-// AddPortToInstanceGroup fakes adding ports to an Instance Group.
-func (f *FakeInstanceGroups) AddPortToInstanceGroup(ig *compute.InstanceGroup, port int64) (*compute.NamedPort, error) {
-	f.Ports = append(f.Ports, port)
-	return &compute.NamedPort{Name: f.namer.BeName(port), Port: port}, nil
+func (f *FakeInstanceGroups) SetNamedPortsOfInstanceGroup(igName, zone string, namedPorts []*compute.NamedPort) error {
+	found := false
+	for _, ig := range f.instanceGroups {
+		if ig.Name == igName && ig.Zone == zone {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return fmt.Errorf("Failed to find instance group %q in zone %q", igName, zone)
+	}
+
+	f.Ports = f.Ports[:0]
```

Comment:
This seems odd to me. Can we change the `FakeInstanceGroup.Ports []int64` to `FakeInstanceGroup.Port int64`?